pilot/pkg/security/authz/converter/converter.go

```diff
@@ -362,7 +362,7 @@ func (c *Converter) v1alpha1ModelTov1beta1Policy(v1alpha1Model *authz_model.Mode
 		}
 		for _, service := range accessRule.Services {
 			for j, selector := range c.getSelectors(service, namespace) {
-				name := fmt.Sprintf("service-%s-%d", strings.ReplaceAll(service, "*", "wildcard"), j)
+				name := fmt.Sprintf("service-%s-rule%d-target%d", strings.ReplaceAll(service, "*", "wildcard"), i, j)
```

Comment:
if there are multiple workload selectors for a service field in v1alpha1 (e.g. prefix service), would it be better if we use the label in the AuthorizationPolicy name (like producepage-app-v1), rather than some index like "target0", which may be harder to know which workload is enforcing this policy?

Reply:
yes, I think we should also include the name of the `ServiceRole` later to make it more clear.